core/src/main/scala/caliban/schema/Schema.scala

```diff
@@ -176,7 +176,8 @@ trait GenericSchema[R] extends SchemaDerivation[R] with TemporalSchema {
     directives: List[Directive] = List.empty
   ): Schema[R1, A] =
     new Schema[R1, A] {
-      override def toType(isInput: Boolean, isSubscription: Boolean): __Type =
+      override def toType(isInput: Boolean, isSubscription: Boolean): __Type = {
+        val _ = resolver
```

Comment:
Why not just making `resolver` a `val`? It needs to be lazy?

Reply:
It needs to be lazy otherwise we get a stack-overflow error as the derivation for the field schemas hasn't been performed yet